src/core/EventDispatcher.js

```diff
@@ -4,18 +4,39 @@
 
 THREE.EventDispatcher = function () {};
 
-THREE.EventDispatcher.prototype = {
+//
+// [Deprecation]
+//
+
+THREE.EventDispatcher.prototype = Object.assign( Object.create( {
```

Comment:
Because I had to pull a little "magic trick" in order to safely deprecate `.apply`:

When it's directly on the prototype it will be `assign`ed to the target prototype. Firstly it doesn't belong there and secondly it may clobber client code due to a name conflict (mixing into a class that happens to have an own method coincidentally called `apply`).

I solved it by setting up a prototype chain. This way it won't be copied.

Also when removing the deprecation, `.constructor` must not be set on the prototype, so mixing it in won't set `.constructor` on the target.

I hope [this comment](https://github.com/tschw/three.js/blob/EventDispatcher/src/core/EventDispatcher.js#L27-L31) is documents it sufficiently.